zerver/webhooks/gitlab/view.py

```diff
@@ -408,6 +410,77 @@ def get_release_event_body(payload: WildValue, include_title: bool) -> str:
     return body
 
 
+def get_emoji_event_transformed_type(payload: WildValue, type: str) -> str:
+    if type == "MergeRequest":
+        return "MR"
+    elif type == "DesignManagement::Design":
+        return "design"
+    elif type == "Note":
+        type = payload["note"]["noteable_type"].tame(check_string)
+        return get_emoji_event_transformed_type(payload, type)
+    return type.lower()
+
+
+def get_emoji_event_subtype_message(type: str) -> str:
+    return "a comment on " if type == "Note" else ""
+
+
+def get_emoji_event_url_id(payload: WildValue, type: str) -> tuple[str, str]:
+    if type == "design":
+        comment_url, _, design_name = parse_design_comment(
+            payload["note"], payload["project"]["web_url"].tame(check_string)
+        )
+        return comment_url, design_name
+
+    url = payload["object_attributes"]["awarded_on_url"].tame(check_string)
+    return url, url.split("/")[-1].split("#")[0]
```

Comment:
Can we add a validation check on the url before this or do a regex? 
I fear this could return an empty string or wrong value if the url doesn't end with a "/" or is malformed for any reason.

Reply:
Let's also add a comment above this how we expect the url to look